cmd/osmosisd/cmd/balances_from_state_export.go

```diff
@@ -315,6 +315,32 @@ Example:
 				snapshotAccs[addr] = account
 			}
 
+			queryClient := authtypes.NewQueryClient(clientCtx)
+
+			for addr, _ := range snapshotAccs {
+
+				acc, err := queryClient.Account(cmd.Context(), &authtypes.QueryAccountRequest{
+					Address: addr,
+				})
+				if err != nil {
+					return err
+				}
+
+				moduleAccount, ok := acc.(*authtypes.ModuleAccount)
+				if !ok {
+					continue
+				}
+
+				moduleAccount, ok = acc.(authtypes.ModuleAccount)
+				if !ok {
+					continue
+				}
```

Comment:
@mattverse These checks are giving me two errors:
1. `moduleAccount is declared but not used`
2. `acc (variable of type *"github.com/cosmos/cosmos-sdk/x/auth/types".QueryAccountResponse) is not an interface`

Also, how is "ok" different from "err"? Should I be deleting from the snapshotAccs map (i.e. if the cast to `*authtypes.ModuleAccount` is successful) in these if statements instead of continuing?

Reply:
we conventionally use ok as a variable to see if type casting is legit or not. 
if !ok, it means that type casting the interface wouldnt work. 
```suggestion
		res, err := queryClient.Account(cmd.Context(), &authtypes.QueryAccountRequest{
					Address: addr,
				})
				if err != nil {
					return err
				}
				var newAccount authtypes.AccountI
				err = clientCtx.InterfaceRegistry.UnpackAny(res.Account, &newAccount)
				if err != nil {
					panic(err)
				}
				
				moduleAccount, ok := newAccount.(*authtypes.ModuleAccount)
				if !ok {
					continue
				}
```

So basically there are two things to get fixed here. 
1. queryClient.Account is not returning an account right away, it returns account wrapped inside a proto struct called `QueryAccountResponse`. Therefore we can't call acc right away, it should be inside the response. 
2. queryClient.Account returns any, so we also have to unpack any, and then use the unpacked any to do whatsoever logic that needs to be done.